vault/expiration.go

```diff
@@ -707,16 +707,12 @@ func (m *ExpirationManager) LazyRevoke(ctx context.Context, leaseID string) erro
 	}
 
 	le.ExpireTime = time.Now()
-	{
-		m.pendingLock.Lock()
-		if err := m.persistEntry(ctx, le); err != nil {
-			m.pendingLock.Unlock()
-			return err
-		}
-
-		m.updatePendingInternal(le)
+	// TODO: potential race with concurrent renew.
+	if err := m.persistEntry(ctx, le); err != nil {
 		m.pendingLock.Unlock()
```

Comment:
i'm not seeing where `m.pendingLock` was `Lock()`d in the new code?

Reply:
You're correct, both these pendingLock calls should be removed.